### PR TITLE
docs: add an FAQ doc

### PR DESCRIPTION
**Pull request by eysi09**
**What this PR does / why we need it**:

Adds a FAQ doc, based on questions from the community. Also adds some entries to the Troubleshooting guide. 

Would be great to get multiple opinions on this so I'm requesting a few reviewers.

### CODE REVIEW

File: docs/misc/faq.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+### How do I annotate ingresses?
+
+You can set annotations on ingresses under the [`services[].ingresses[]` field](https://docs.garden.io/reference/module-types/container#services-ingresses-annotations).
+
+### What versions and variations of Kubernetes does Garden support?
+
+Garden interfaces with your cluster via `kubectl` and by using the Kubernetes APIs directly and should therefore work with all Kubernetes clusters that implement these. Garden is committed to supporting the latest six stable versions of Kubernetes.
+
+### Can I add Kubernetes specific fields to the `container` module (e.g. annotations and labels)?
```
Comment (thsig) [inline docs/misc/faq.md]: "Kubernetes specific" -> "Kubernetes-specific"

"to the `container` module" -> "to `container` modules"

File: docs/misc/faq.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+No, you have to use the [`kubernetes`](https://docs.garden.io/reference/module-types/kubernetes) module for that.
```
Comment (thsig) [inline docs/misc/faq.md]: "module for that" -> "module type for that"

File: docs/misc/faq.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+### When are you releasing the Plugin SDK?
+
+We're exploring how we can release it incrementally. Please let us know if this is something you're interested in.
+
+### What system components does Garden install?
+
+The components installed when using the remote building functionality are discussed in the [In-cluster building docs](https://docs.garden.io/using-garden/in-cluster-building).
+
+Garden also optionally installs Nginx. The `local-kubernetes` provider defaults to installing Nginx but the (remote) `kubernetes` provider does not install it by default.
```
Comment (thsig) [inline docs/misc/faq.md]: "installing Nginx but" -> "installing Nginx, but"

File: docs/misc/faq.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+Furthermore, the `openfaas` provider installs some components necessary for OpenFaas to work.
+
+We of course use Garden to install these components and you’ll find the Garden modules for them in [in our source code](https://github.com/garden-io/garden/tree/master/garden-service/static) under `kubernetes/system` and `openfaas/system`.
```
Comment (thsig) [inline docs/misc/faq.md]: "We of course use Garden" -> "Of course, we use Garden"

"these components and" -> "these components, and"

File: docs/misc/faq.md [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+
+### How does Garden resolve the `*.local.app.garden` domain?
+
+The `*.local.app.garden` domain resolves to 127.0.0.1 via our DNS provider for convenience. If you want to use a different hostname for local development you’ll have to add the corresponding entry to your hosts file.
```
Comment (thsig) [inline docs/misc/faq.md]: "development you'll" -> "development, you'll"

File: docs/misc/faq.md [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+
+### Does garden support bi-directional syncing?
+
+No, it doesn't. See [this question](#how-do-i-access-files-that-are-generated-at-runtime-eg-migration-files-that-are-checked-into-version-control) above for accessing files that are generated at runtime.
+
+### Is Garden stable or should I wait for 1.0?
+
+Garden is currently in use by many teams. We don’t have a set date or plan to label it as 1.0, but we don't expect to do it anytime soon. For comparison, very widely used tools like Terraform are still not at 1.0.
+
+We have a team of people working on it full time and we make it a priority to address all non-trivial bugs. We’re also happy to help out and answer questions via our community Slack.
```
Comment (thsig) [inline docs/misc/faq.md]: "full and time" -> "full-time, and"

File: docs/misc/faq.md [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+
+### Does Garden work offline?
+
+Garden is not currently designed to work in air-gapped environments, and would require a fair amount of workarounds, unfortunately.
```
Comment (thsig) [inline docs/misc/faq.md]: "environments, and would require" -> "environments. This would require"

File: docs/misc/troubleshooting.md [hunk 8]
```diff
@@ -1,27 +1,73 @@
...
+
+There's an [open pull request](https://github.com/helm/helm/pull/7653) for a fix.
+
+### Files are missing from build context
+
+This is likely because they're being excluded somewhere, e.g. in `.gitignore` or `.gardenignore`. Garden currently respects `.gitignore` but we plan to change that in our next major release.
+
+### `ErrImagePull` when referencing an image from a `container` module in a `helm` module
+
+Make sure to use the `output` from the respective container module.
```
Comment (thsig) [inline docs/misc/troubleshooting.md]: "the `output` from the respective container module" -> "the `outputs` field from the `container` module being referenced"

File: docs/misc/troubleshooting.md [hunk 9]
```diff
@@ -1,27 +1,73 @@
...
+
+Do:
```
Comment (thsig) [inline docs/misc/troubleshooting.md]: "Do:" -> "For example:"

**Comment by eysi09** [comment]
I've addressed the comments. Anything else that needs improving, @thsig, @edvald?

**Pull request merged**